src/NUnitFramework/framework/Internal/Execution/TextMessageWriter.cs

```diff
@@ -164,41 +164,43 @@ public override void DisplayDifferences(object? expected, object? actual, Tolera
             }
         }
 
-        /// <summary>
-        /// Display the expected and actual string values on separate lines.
-        /// If the mismatch parameter is >=0, an additional line is displayed
-        /// line containing a caret that points to the mismatch point.
-        /// </summary>
-        /// <param name="expected">The expected string value</param>
-        /// <param name="actual">The actual string value</param>
-        /// <param name="mismatch">The point at which the strings don't match or -1</param>
-        /// <param name="ignoreCase">If true, case is ignored in string comparisons</param>
-        /// <param name="clipping">If true, clip the strings to fit the max line length</param>
-        public override void DisplayStringDifferences(string expected, string actual, int mismatch, bool ignoreCase, bool clipping)
+        /// <inheritdoc/>
+        public override void DisplayStringDifferences(string expected, string actual, int mismatchExpected, int mismatchActual, bool ignoreCase, bool ignoreWhiteSpace, bool clipping)
         {
             // Maximum string we can display without truncating
             int maxDisplayLength = MaxLineLength
                 - PrefixLength // Allow for prefix
                 - 2;           // 2 quotation marks
 
             if (clipping)
-                MsgUtils.ClipExpectedAndActual(ref expected, ref actual, maxDisplayLength, mismatch);
-
-            expected = MsgUtils.EscapeControlChars(expected);
-            actual = MsgUtils.EscapeControlChars(actual);
+            {
+                if (ignoreWhiteSpace)
+                {
+                    expected = MsgUtils.ClipWhenNeeded(expected, expected.Length, maxDisplayLength, ref mismatchExpected);
+                    actual = MsgUtils.ClipWhenNeeded(actual, actual.Length, maxDisplayLength, ref mismatchActual);
+                }
+                else
+                {
+                    MsgUtils.ClipExpectedAndActual(ref expected, ref actual, maxDisplayLength, ref mismatchExpected, ref mismatchActual);
+                }
+            }
 
-            // The mismatch position may have changed due to clipping or white space conversion
-            mismatch = MsgUtils.FindMismatchPosition(expected, actual, 0, ignoreCase);
+            expected = MsgUtils.EscapeControlChars(expected, ref mismatchExpected);
+            actual = MsgUtils.EscapeControlChars(actual, ref mismatchActual);
 
             Write(Pfx_Expected);
             Write(MsgUtils.FormatValue(expected));
             if (ignoreCase)
                 Write(", ignoring case");
+            if (ignoreWhiteSpace)
+                Write(", ignoring white-space");
             WriteLine();
+            if (mismatchExpected >= 0 && mismatchExpected != mismatchActual)
+                WriteCaretLine(mismatchExpected);
             WriteActualLine(actual);
             //DisplayDifferences(expected, actual);
```

Comment:
Should we remove this line?

Reply:
I presume you mean the one commented out and not the one I just added?

I suspect this line was commented out when the `WriteCaretLine` was added, but all of this was already there in the first commit to the repo 15 years ago. I agree it should be removed.